model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py

```diff
@@ -1025,6 +1040,192 @@ async def execute(self, user: User, model_endpoint_name: str) -> GetLLMModelEndp
+            raise EndpointInfraStateNotFound(error_msg)
+
+        infra_state = model_endpoint.infra_state
+
+        if (
+            request.endpoint_type is not None
+            and request.endpoint_type != endpoint_record.endpoint_type
+        ):
+            raise ObjectHasInvalidValueException(
+                f"Cannot change endpoint type (attempting to change from {endpoint_record.endpoint_type} to {request.endpoint_type})"
+            )
+
+        if (
+            request.model_name
+            or request.source
+            or request.inference_framework
+            or request.inference_framework_image_tag
+            or request.num_shards
+            or request.quantize
+            or request.checkpoint_path
+        ):
+            llm_metadata = (model_endpoint.record.metadata or {}).get("_llm", {})
+
+            inference_framework = llm_metadata["inference_framework"]
+            # disallowing changing the inference framework for now since this might cause downtime
+            # (for completion requests, gateway determines the payload keys sent to the user container based on the framework)
+            if (
+                request.inference_framework is not None
+                and request.inference_framework != inference_framework
+            ):
+                raise ObjectHasInvalidValueException(
+                    f"Cannot change inference framework (attempting to change from {inference_framework} to {request.inference_framework})"
+                )
+
+            model_name = request.model_name or llm_metadata["model_name"]
+            source = request.source or llm_metadata["source"]
+            inference_framework_image_tag = (
+                request.inference_framework_image_tag
+                or llm_metadata["inference_framework_image_tag"]
+            )
+            num_shards = request.num_shards or llm_metadata["num_shards"]
+            quantize = request.quantize or llm_metadata.get("quantize")
+            checkpoint_path = request.checkpoint_path or llm_metadata.get("checkpoint_path")
+
+            validate_model_name(model_name, inference_framework)
+            validate_num_shards(
+                num_shards, inference_framework, request.gpus or infra_state.resource_state.gpus
+            )
+            validate_quantization(quantize, inference_framework)
+
+            if inference_framework in [
+                LLMInferenceFramework.TEXT_GENERATION_INFERENCE,
+                LLMInferenceFramework.VLLM,
+                LLMInferenceFramework.LIGHTLLM,
+                LLMInferenceFramework.TENSORRT_LLM,
+            ]:
+                if endpoint_record.endpoint_type != ModelEndpointType.STREAMING:
+                    raise ObjectHasInvalidValueException(
+                        f"Creating endpoint type {str(endpoint_record.endpoint_type)} is not allowed. Can only create streaming endpoints for text-generation-inference, vLLM and LightLLM."
```

Comment:
also should we mention tensorrtllm in the error as well

Reply:
Yeah we probably want a shared constant string or helper function to use in both error sites.